modules/caddy-automatic-https/locals.tf

```diff
@@ -0,0 +1,34 @@
+locals {
+  # Get a name from the descriptor. If not available, use default naming convention.
+  # Trim and replace function are used to avoid bare delimiters on both ends of the name and situation of adjacent delimiters.
+  name_from_descriptor = trim(replace(
+    lookup(module.this.descriptors, "azure-container-group", module.this.id),
+    "/${module.this.delimiter}${module.this.delimiter}+/",
+  module.this.delimiter), module.this.delimiter)
+
+  location            = coalesce(one(data.azurerm_resource_group.this[*].location), var.location)
+  resource_group_name = coalesce(one(data.azurerm_resource_group.this[*].name), var.resource_group_name)
+
+  dns_name_label = (length(var.subnet_ids) == 0
+    ? (var.dns_name_label != null ? var.dns_name_label : local.name_from_descriptor)
```

Comment:
Minor - I don;t like that linting - `?` in the new line
Maybe few more new lined would make it more clear